matrix/client/endpoint_cache.py

```diff
@@ -36,24 +36,35 @@ def __init__(
     async def __call__(self, force_update=False):
         if time.time() - (self.timestamp or 0) > self.ttl or force_update:
             async with self.lock:
-                if time.time() - (self.timestamp or 0) > self.ttl or force_update:
-                    try:
-                        if self.serve_app:
-                            self.timestamp = time.time()
-                            status, content = await fetch_url(
-                                self.ray_address + "/api/serve/applications/",
-                                headers={"Accept": "application/json"},
+                try:
+                    if self.serve_app:
+                        self.timestamp = time.time()
+                        status, content = await fetch_url(
+                            self.ray_address + "/api/serve/applications/",
+                            headers={"Accept": "application/json"},
+                        )
+                        if status is not None and status == 200:
+                            ray_query_result = json.loads(content)
+                            # note here we did not loop deployment, assuming there is only one deployment (but multiple replicas) for an app
+                            self.ips = set(
+                                [
+                                    replica["node_ip"]
+                                    for replica in next(
+                                        iter(
+                                            ray_query_result["applications"][
+                                                self.app_name
```

Comment:
I think the proxy is not app specific, for example, if you send a 8B request to a proxy running 405B model, it will proxy correctly.